vermouth/rcsu/contact_map.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2023 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Read RCSU Go model contact maps.
+"""
+
+def read_go_map(file_path, header_lines=0, cols=[2, 6, 10]):
+    """
+    Read a RCSU contact map.
+    """
+    with open(file_path, "r", encoding='UTF-8') as _file:
+        lines = _file.readlines()
```

Comment:
specifying the encoding is now recommended

Reply:
Learn something new every day :)
Do you have a source/reference for this? I can't find it in the docs.

Reply:
pylint complains a LOT about this. I've seen it in many other major packages as well but couldn't find it in PEP so far.

Reply:
Why not? It's just the way I've written all file readers. I think follows a pattern by Jon

Reply:
It only makes sense if you need to index the lines explicitely, for example to look at the previous or next line. The reason not to do it is because it can be dramatically bad for performance, especially if you have (really) big files. Better to let Python stream the file into memory bit by bit in the background.